pkg/providers/cloudstack/cloudstack.go

```diff
@@ -741,6 +753,10 @@ func buildTemplateMapMD(clusterSpec *cluster.Spec, datacenterConfigSpec v1alpha1
 
 		// Add no-proxy defaults
 		noProxyList = append(noProxyList, common.NoProxyDefaults...)
+		cloudStackManagementApiEndpointHostname, err := getHostnameFromUrl(datacenterConfigSpec.ManagementApiEndpoint)
+		if err == nil {
```

Comment:
don't we need to handle the error case?

Reply:
right, this is more context aware defaults/validations.. if it's too much work, im ok with creating an issue to follow up